lib/ansible/modules/cloud/vmware/vmware_drs.py

```diff
@@ -0,0 +1,279 @@
+  drs_rule_name:
+    description:
+    - The name of the DRS rule to manage.
+    required: True
+  enabled:
+    description:
+    - If set to C(True), the DRS rule will be enabled.
+    - Effective only if C(state) is set to C(present).
+    default: False
+  mandatory:
+    description:
+    - If set to C(True), the DRS rule will be mandatory.
+    - Effective only if C(state) is set to C(present).
+    default: False
+  affinity_rule:
+    description:
+    - If set to C(True), the DRS rule will be an Affinity rule.
+    - If set to C(False), the DRS rule will be an Anti-Affinity rule.
+    - Effective only if C(state) is set to C(present).
+    default: False
+extends_documentation_fragment: vmware.documentation
+'''
+
+EXAMPLES = r'''
+- name: Create DRS Affinity Rule for VM-VM
+  vmware_drs:
+    hostname: "{{ esxi }}"
+    username: "{{ esxi_username }}"
+    password: "{{ esxi_password }}"
+    cluster: "{{ cluster_name }}"
+    validate_certs: no
+    vms:
+        - vm1
+        - vm2
+    drs_rule_name: vm1-vm2-affinity-rule-001
+    enabled: True
+    mandatory: True
+    affinity_rule: True
+
+- name: Create DRS Anti-Affinity Rule for VM-VM
+  vmware_drs:
+    hostname: "{{ esxi }}"
+    username: "{{ esxi_username }}"
+    password: "{{ esxi_password }}"
+    cluster: "{{ cluster_name }}"
+    validate_certs: no
+    vms:
+        - vm1
+        - vm2
+    drs_rule_name: vm1-vm2-affinity-rule-001
+    enabled: True
+    mandatory: True
+    affinity_rule: False
+
+- name: Delete DRS Affinity Rule for VM-VM
+  vmware_drs:
+    hostname: "{{ esxi }}"
+    username: "{{ esxi_username }}"
+    password: "{{ esxi_password }}"
+    cluster: "{{ cluster_name }}"
+    validate_certs: no
+    drs_rule_name: vm1-vm2-affinity-rule-001
+    state: absent
+'''
+
+RETURN = r'''#'''
+
+try:
+    from pyVmomi import vim, vmodl
+except ImportError:
+    pass
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.vmware import (PyVmomi, vmware_argument_spec, wait_for_task,
+                                         find_vm_by_id, find_cluster_by_name)
+
+
+class VmwareDrs(PyVmomi):
+    def __init__(self, module):
+        super(VmwareDrs, self).__init__(module)
+        self.vm_list = module.params['vms']
+        self.cluster_name = module.params['cluster']
+        self.rule_name = module.params['drs_rule_name']
+        self.enabled = module.params['enabled']
+        self.mandatory = module.params['mandatory']
+        self.affinity_rule = module.params['affinity_rule']
+
+        # Sanity check for cluster
+        self.cluster_obj = find_cluster_by_name(content=self.content,
+                                                cluster_name=self.cluster_name)
+        self.vm_obj_list = []
+        if self.cluster_obj is None:
+            self.module.fail_json(msg="Failed to find the cluster %s" % self.cluster_name)
+        # Sanity check for virtual machines
+        self.get_all_vms_info()
```

Comment:
Maybe you should set `self.vm_obj_list = []` between line 133 and 134 (after the comment "Sanity check for virtual machines")

Reply:
Any specific reason ?

Reply:
`self.vm_obj_list = []` doesn't have to do with `# Sanity check for cluster` but with `# Sanity check for virtual machines`.
It's just for the code to be more coherent with the comments.